sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SymbolicExecution/SymbolicValues/SymbolicValue.cs

```diff
@@ -319,8 +214,8 @@ protected IEnumerable<ProgramState> ThrowIfTooMany(IEnumerable<ProgramState> sta
             return programStates;
         }
 
-        private IEnumerable<ProgramState> TrySetConstraint(BoolConstraint boolConstraint,
-            SymbolicValueConstraints oldConstraints, ProgramState currentProgramState)
+        private IEnumerable<ProgramState> TrySetBoolConstraint(BoolConstraint constraint,

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'SonarAnalyzer.SymbolicExecution.SymbolicValueConstraint' instead of 'SonarAnalyzer.SymbolicExecution.Constraints.BoolConstraint'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS3242)